File: Python/kinCost.py
import scipy as sp
import numpy as np

def kinCost(x, *data):
    P, ConsMat, LockMat, distVec = data
    P[~LockMat] = x
    dMat = sp.spatial.distance.squareform(sp.spatial.distance.pdist(np.transpose(P)))
    dVec = dMat[ConsMat]
    return (dVec - distVec)**2

File: MATLAB/kinCost.m
function err = kinCost(x, P, ConsMat,LockMat,distVec)

%1)update hardpoint matrix
P(~LockMat) = x(:);

%2) calculate new distances from constraint equations
dMat = squareform(pdist(P'));
dVec = dMat(ConsMat);

%3) calculate errors
err = (dVec - distVec).^2;

end
